tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -55,10 +64,21 @@ size_t http_parser_execute (http_parser *parser,
   __CPROVER_assert(data, "http_parser_execute data nonnull");
 
   _httpsResponse_t *_httpsResponse = (_httpsResponse_t *)(parser->data);
-  /* nondet_bool is required to get coverage. */
+  // Choose whether the parser found the header
   _httpsResponse->foundHeaderField = nondet_bool();
   _httpsResponse->parserState = PARSER_STATE_BODY_COMPLETE;
 
+  // Generate the header value found
+  if (_httpsResponse->foundHeaderField) {
+    size_t valueLen;
+    __CPROVER_assume(valueLen <= len);
+    _httpsResponse->pReadHeaderValue = malloc(valueLen+1);
+    _httpsResponse->pReadHeaderValue[valueLen] = 0;
+    _httpsResponse->readHeaderValueLength = valueLen;
+  }
+
+  // ???: Should this be valueLen?
+  // ???: Does it suggest a problem that returning any length works?
```

Comment:
This should not be valueLen, the amount parsed will always be <= len. So the return value of this routine is <= len.

Reply:
ValueLen is constrained to be between 0 <= valueLen <= len.   So that is the value you are asking for, and I will make that change.

Reply:
I'm changing the name to valueLength.  I think the name valueLen (a name you use a parameter to other functions) was causing confusion in this discussion.